src/xo-app/vm/tab-disks.js

```diff
@@ -485,103 +607,22 @@ export default class TabDisks extends Component {
-                        labelProp='name_label'
-                        value={sr}
-                        useLongClick
-                      >
-                        <Link to={`/srs/${sr.id}`}>{sr.name_label}</Link>
-                      </XoSelect>
-                    }
-                    </td>
-                    {vm.virtualizationMode === 'pv' && <td>
-                      <Toggle
-                        value={vbd.bootable}
-                        onChange={bootable => setBootableVbd(vbd, bootable)}
-                      />
-                    </td>}
-                    <td>
-                      <StateButton
-                        disabledLabel={_('vbdStatusDisconnected')}
-                        disabledHandler={connectVbd}
-                        disabledTooltip={_('vbdConnect')}
-
-                        enabledLabel={_('vbdStatusConnected')}
-                        enabledHandler={disconnectVbd}
-                        enabledTooltip={_('vbdDisconnect')}
-
-                        disabled={!(vbd.attached || isVmRunning(vm))}
-                        handlerParam={vbd}
-                        state={vbd.attached}
-                      />
-                    </td>
-                    <td className='text-xs-right'>
-                      <Tooltip content={_('vdiMigrate')}>
-                        <ActionRowButton
-                          icon='vdi-migrate'
-                          handler={this._migrateVdi}
-                          handlerParam={vdi}
-                        />
-                      </Tooltip>
-                      {!vbd.attached &&
-                        <span>
-                          <Tooltip content={_('vdiForget')}>
-                            <ActionRowButton
-                              icon='vdi-forget'
-                              handler={deleteVbd}
-                              handlerParam={vbd}
-                            />
-                          </Tooltip>
-                          <Tooltip content={_('vdiRemove')}>
-                            <ActionRowButton
-                              icon='vdi-remove'
-                              handler={deleteVdi}
-                              handlerParam={vdi}
-                            />
-                          </Tooltip>
-                        </span>
-                      }
-                    </td>
-                  </tr>
-                })}
-              </tbody>
-            </table>
+        foo
+          {!isEmpty(vdis)
+             ? ((vm.virtualizationMode === 'pv') ? <SortedTable
+               collection={vdis}
+               columns={COLUMNS}
+               individualActions={INDIVIDUAL_ACTIONS}
+               userData={userdata}
+                /> : <SortedTable
+                  collection={vdis}
+                  columns={COLUMNS2}
+                  individualActions={INDIVIDUAL_ACTIONS}
+                  userData={userdata}
+                />)
```

Comment:
No need to duplicate the whole `SortedTable` if the only difference is `columns`.